asv/plugins/virtualenv.py

```diff
@@ -108,24 +97,30 @@ def name(self):
 
     @classmethod
     def matches(self, python):
-        if not (re.match(r'^[0-9].*$', python) or re.match(r'^pypy[0-9.]*$', python)):
+        match = re.match(r'\A(?:pypy)?([0-9.]*)\Z', python)
+        if not match:
             # The python name should be a version number, or pypy+number
             return False
+        executable = Virtualenv._find_python(python)
 
-        try:
-            import virtualenv
-        except ImportError:
-            return False
+        if LooseVersion(match.group(1)) < LooseVersion("3.3"):
+            self._virtualenv_argv = [
+                executable, "-mvirtualenv", "--no-site-packages"]
```

Comment:
The point I was making in https://github.com/spacetelescope/asv/issues/523#issuecomment-310868051 is that venv is unable to create virtualenvs for pythons others than the current interpreter, so if we want the master python to use venv, then any other python in the test matrix either needs to be >=3.3 (i.e., have venv itself), or have virtualenv installed.

As I only work with Py3 that restriction seems acceptable to me, but I can also understand if you disagree, in which case I don't think such an approach can ever be made to work (unless builtin venv picks up the ability to create venvs for other pythons, but why whould they?).  So, please let me know if I should try to fix the tests, or if you simply won't accept such a design (again, it's up to you, of course).